tareas/rafaelval/rvcoria_Tarea-1.jmd

```diff
@@ -0,0 +1,286 @@
+Base.:*(u::Real,v::Dual) = *(promote(u,v)...)
+
+Base.:/(u::Dual,v::Real) = /(promote(u,v)...)
+Base.:/(u::Real,v::Dual) = /(promote(u,v)...)
+```
+
+Para probar que los resultados sean correctos: 
+   \begin{eqnarray}
+       x &=& 1 \\
+       \overleftrightarrow{y} &=& 1 + 2\epsilon
+   \end{eqnarray} 
+Con lo que debería obtener:
+    \begin{eqnarray}
+        \overleftrightarrow{x} + \overleftrightarrow{y}&=& (1+1) + (2+0)\epsilon  &=& 2 + 2\epsilon &=& (2,2) \\
+        \overleftrightarrow{x} - \overleftrightarrow{y} &=& (1-1) + (0-2)\epsilon  &=& 0 - 2\epsilon  &=& (0,-2)\\
+        \overleftrightarrow{x} \cdot \overleftrightarrow{y} &=&  (1\cdot1) + (2\cdot1+1\cdot0)\epsilon &=& 1 + 2\epsilon  &=& (1,2)\\
+        \overleftrightarrow{x}/\overleftrightarrow{y} &=& (1/1) + \left(\frac{0\cdot1-2\cdot1}{1^2}\right)\epsilon &=& 1 - 2\epsilon &=& (1,-2)
+       \end{eqnarray}
+
+```julia
+x = 1
+y = Dual(1.0,2.0)
+println("x+y: ",isequal(x+y, Dual(2,2)) )
+println("x-y: ",isequal(x-y, Dual(0,-2)) )
+println("x*y: ",isequal(x*y, Dual(1,2)) )
+println("x/y: ",isequal(x/y, Dual(1,-2)) )
+```
+
+# Ejercicio 3
+Definan una nueva función `var_dual(x_0)` cuyo resultado sea un `Dual` cuya parte principal es `x_0` y cuya parte derivada sea tal que `var_dual(x_0)` *represente* a la variable independiente `x` en `x_0`. La idea es piensen qué debe representar la parte derivada del dual `xd` que regresa `xd = var_dual(x_0)`. Con esta función evalúen `g(xd)`
+donde
+
+$$
+g(x) = \frac{3x^2-8x+1}{7x^3-1}.
+$$
+
+Según la interpretación dada, esta función puede calcularse como sigue:
+$$g(x+\epsilon x^\prime) = g(x) + \epsilon x^\prime\frac{dg(x)}{dx} $$
+
+donde 
+$$g(x) = \frac{3x^2-8x+ 1}{7x^3 - 1}$$
+y 
+$$\frac{dg(x)}{dx} = \frac{(6x-8)(7x^3-1)-(21x^2)(3x^2 - 8x + 1)}{(7x^3 - 1)^2}$$
+
+
+```julia
+#Por la interpretación dada, para que la parte var_dual(x_0) represente a la variable independiente x, 
+#la segunda entrada del dual debe ser la derivada de la identidad, es decir, 1.
+
+var_dual(x_0::T) where {T<:Real} = Dual(x_0,one(x_0))
+```
+
+```julia
+#Defino a la función g(x) de tal manera que acepte duales:
+function g(a::Dual)
+    u = 3*(a*a) -8*(a) + 1
+    v = 7*(a*(a*a)) - 1
+    return u/v
+end
+
+#Ahora puedo evaluar a g en xd = var_dual(x_0). Usaré un punto en particular 
+
+xd = var_dual(2.0)
+println("g(xd) = ", g(xd))
+
+# Por la interpretación dada, este valor debería coincidir con (g(x_0),g'(x_0)) ya que la segunda entrada del 
+# zdual es 1
+#Para probar esto calcularé a la funció usando esta última expresión:
+
+function g1(a::Dual) #Es la misma función g pero esta sólo considera la primer entrada del dual
+    u = 3*(a.x)^2 -8*(a.x) + 1
+    v = 7*(a.x)^3 - 1
+    return u/v
+end
+
+function dg1(a::Dual)  #Derivada de g
+    h = (6*(a.x)-8)*(7*(a.x)^3 - 1)-(21*(a.x)^2)*(3*(a.x)^2 - 8*(a.x) + 1)
+    l = (7*(a.x)^3 - 1)^2
+    return h/l
+end
+
+g2(a::Dual) = Dual(g1(a),one(a.x2)*dg1(a)...) ##Calcula usando la interpretación
```

Comment:
Los `...` no son necesarios aquí.

Reply:
Luis, estoy tratando de subir la actualización con estos  comentarios ya corregidos, pero no sé porqué no aparece. Cuando uso "git push mifor csga/T1" me devuelve "Everything up-to-date" y al abrir git en el navegador no me aparece que haya actualizado mi tarea1

Reply:
Checa que el archivo .jmd que generas a partir del .ipynb corresponda con el que subes, es decir `tareas/rafaelval/rvcoria_Tarea-1.jmd`. Recuerda también que debes hacer `git add` y  `git commit`